sdk/identity/identity/src/credentials/azureCliCredential.ts

```diff
@@ -183,4 +180,35 @@ export class AzureCliCredential implements TokenCredential {
       }
     });
   }
+
+  /**
+   * Parses the raw JSON response from the Azure CLI into a usable AccessToken object
+   *
+   * @param rawResponse - The raw JSON response from the Azure CLI
+   * @returns An access token with the expiry time parsed from the raw response
+   *
+   * The expiryTime of the credential's access token, in milliseconds, is calculated as follows:
+   *
+   * When available, expires_on (introduced in Azure CLI v2.54.0) will be preferred. Otherwise falls back to expiresOn.
+   */
+  private parseRawResponse(rawResponse: string): AccessToken {
+    const response: any = JSON.parse(rawResponse);
+    const token = response.accessToken;
+    // if available, expires_on will be a number representing seconds since epoch.
+    // ensure it's a number or NaN
+    const expiresOnTimestamp = Number.parseInt(response.expires_on) * 1000;
+    if (!isNaN(expiresOnTimestamp)) {
+      logger.getToken.info("expires_on is available and is valid, using it");
+      return {
+        token,
+        expiresOnTimestamp,
+      };
+    }
+
+    // fallback to the older expiresOn - an RFC3339 date string
+    return {
+      token,
+      expiresOnTimestamp: new Date(response.expiresOn).getTime(),
```

Comment:
what if this throws? we bubble it up?

Reply:
I think it would bubble up - or even worse maybe return NaN? The reason I left it as such is because that was the existing behavior and if I was to change it, what would I change it to?

I can add some logging though, if that would help? Or a more helpful error message?

Reply:
I did end up guarding and throwing an error in the event that we get an invalid value 👍